tgui/packages/tgui/interfaces/Manufacturer/index.tsx

```diff
@@ -0,0 +1,379 @@
+
+/* This code block explains the general flow of the code regarding menu structure
+   This menu is rather big so hopefully it helps navigate the heavy nesting of elements
+
+Window contains:
+- Left section containing collapsibles of buttons
+- Right section containing settings
+
+  Left section contains:
+    - Collapsibles with buttons
+
+    Collapsibles contain:
+      - Buttons which have image/name/tooltip based off own data.
+
+    Buttons contain:
+      - Leftmost: image of item
+      - Middle: vertical stack, settings button on top and ? tooltip on bottom.
+        - settings tooltip NYI
+        - tooltip button contains information on material and time costs
+      - Right: name, up to 3 lines / 30 characters
+
+  Right section contains:
+    - Search bar
+    - Materials list containing material info/control
+
+    Material list elements contain:
+      - Eject button
+      - Priority swap buttons (swap priority of two materials for fabricator use)
+      - Speed control buttons
+      - Repeat toggle button
+      - Ore buying which contains ores by rockbox
+
+      Each rockbox element contains:
+        - List of ores available for purchase
+
+        Each ore available for purchase contains:
+          - Button with ore icon, name, amount, and cost
+*/
+
+const BlueprintTooltip = (props) => {
+  let { blueprint } = props;
+  let requirements = [];
+  for (let i in blueprint.item_names) {
+    requirements.push(<>{blueprint.item_amounts[i]/10} {blueprint.item_names[i]}<br /></>);
+  }
+  return (
+    <>
+      Resource Costs:<br />
+      {requirements} <br />
+      {formatTime(blueprint.time, "Instananeous")}
+    </>
+  );
+};
+
+const BlueprintButton = (props, context) => {
+  const { act } = useBackend(context);
+  const { name, blueprintData } = props;
+
+  return (
+    <Stack width={15} inline backgroundColor={backgroundPop} mx={0.575} mb={0} p={0.5}>
+      <Stack.Item>
+        <Button
+          ellipsis
+          width={12.2}
+          height={5.3}
+          p={0}
+          onClick={() => act("product", { "blueprint_ref": blueprintData.ref })}
+        >
+          <Stack>
+            <Stack.Item
+              ml={0}
+              pt={0}
+              mt={0}
+              backgroundColor={backgroundPop}
+            >
+              <Image pixelated src={blueprintData.img} width={5} />
+            </Stack.Item>
+            <Stack.Item>
+              <Stack
+                vertical
+                fill
+                style={{ "align-items": "center" }}
+              >
+                <Stack.Item
+                  width={6}
+                  textAlign="center"
+                  style={{ "white-space": "normal" }}
+                >
+                  {truncate(name, 30)}
+                </Stack.Item>
+              </Stack>
+            </Stack.Item>
+          </Stack>
+        </Button>
+      </Stack.Item>
+
+      <Stack.Item ml={0.5}>
```

Comment:
`Stack.Item`s should have sensible padding as-is. If you're overriding that, maybe you want `Flex` (which is basically a `Stack` without space between things)?